### PR TITLE
[#2145] Dockerisation of RepoSense

### PR DESCRIPTION
**Pull request by asdfghjkxd**
Part of #2145 

### Proposed commit message

```
Currently, there are some issues with regard to consistent testing of
code, especially for the frontend code, due to the difference in
environments where the test cases are run (developer's machine VS
Github Action's workers). This inconsistency in testing results makes
it hard for us to really distinguish false negatives/false positives.

Moreover, this inconsistency in the environment where RepoSense is
installed may also result in dependency issues.

In this PR, a proof-of-concept Dockerfile and docker-compose.yml file
will be provided first to test the viability of such an integration
into RepoSense. Future implementations would see further integration of
Docker into Github Actions, more complicated Docker build processes and
tests and documentation on how to maintain these Docker files and
builds.

Let's move to integrate Docker and Docker Compose into RepoSense to
mitigate and potentially solve some of the aforementioned issues, by
providing users and developers a more standardised environment where
they can run or test RepoSense.
```

### Other information

This PR is part of the following steps outlined in the parent issue:

- [X] Create Ubuntu Dockerfiles
- [X] Integrate Ubuntu Dockerfiles with Github Actions

For this PR, only Ubuntu/Linux images are explored, as the rest are in development. Ubuntu/Linux images are also easier to work with and far less fussy compared to Windows and MacOS images, hence starting work on Ubuntu/Linux might be a helpful proof of concept for the viability of Docker for use for testing.

### CODE REVIEW

**Comment by asdfghjkxd** [comment]
Do note that the Docker containers currently do not conduct frontend tests using cypress, even though the dependencies are installed on the base Ubuntu containers, as they are consistently failing CI. Will look into the cause and update this thread once I have isolated the cause of the error.

File: docker/Dockerfile.ubuntu2204 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#     - https://stackoverflow.com/questions/40873165/use-docker-run-command-to-pass-arguments-to-cmd-in-dockerfile
+# - Installation of Java on Ubuntu
+#     - https://stackoverflow.com/questions/49914574/install-jdk-8-update-172-in-dockerfile-with-ubuntu-image
+# - Installation of NodeJS via Node Version Manager (NVM)
+#     - https://stackoverflow.com/questions/36399848/install-node-in-dockerfile
+
+FROM asdfghjklxd/reposense-images:ubuntu-22-04 AS setup-env
+
+LABEL os=ubuntu
+LABEL "os.version"="20.04"
```
Comment (sopa301) [inline docker/Dockerfile.ubuntu2204:26]: This should probably be 22.04.
```suggestion
LABEL "os.version"="22.04"
```

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout V4
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Build Docker Image
+        run: |
+          docker build -t ${{ matrix.tag }} -f ${{ matrix.dockerfile }} -o ${{ env.OUTDIR }} .
+      # - name: Upload Code Coverage (Java 8)
```
Comment (sopa301) [inline .github/workflows/docker.yml:36]: Is this going to be re-added?

File: docker-compose.yaml
Comment (sopa301) [inline docker-compose.yaml:1]: As discussed, since we're not using this for the CI, we can probably put this in a folder marked for local testing.

**Review by sopa301 — CHANGES_REQUESTED** [review]
LGTM! We just need to fix the Cypress issues.

**Comment by asdfghjkxd** [comment]
As mentioned in issue #2145, we are not pursuing this idea, hence I am closing out this PR.

**Pull request closed without merging**